management_interface/management_interface/test_views.py

```diff
@@ -47,3 +51,73 @@ def test_search_missing_param_returns_bad_request(self):
         self.assertFailure(response, HTTPStatus.BAD_REQUEST, "required")
+
+
+class AdminCareProviderLocationTests(TestCase):
+    def _convert_messages_to_str(self, response):
+        return "".join([str(message) for message in list(response.context["messages"])])
+
+    def _get_upload_file_response(self, csv_file):
+        response = self.client.post(
+            reverse("admin:import_care_recipients", args=(self.location.id,)), {"csvfile": csv_file}, follow=True
+        )
+        return response
+
+    def setUp(self) -> None:
+        self.manager = RegisteredManager.objects.create(
+            given_name="Jehosephat", family_name="McGibbons", cqc_registered_manager_id="My CQC RegsiteredManagerID"
+        )
+        self.location = self.manager.careproviderlocation_set.create(
+            name="My Location Name",
+            email="nosuchaddress@nhs.net",
+            ods_code="My Ods Code",
+            cqc_location_id="My CQC Location ID",
+        )
+
+        self.client = Client()
+        self.user = User.objects.create_user(
+            username="admin", email="admin@example.com", password="password", is_staff=True, is_superuser=True
+        )
+        self.client.login(username="admin", password="password")
+
+    def test_admin_upload_empty_csv_file(self):
+        response = self.client.post(reverse("admin:import_care_recipients", args=(self.location.id,)), follow=True)
+        messages = self._convert_messages_to_str(response)
+        self.assertIn(CSVImportMessages.INVALID_OR_EMPTY_FILE.value, messages)
+        care_recipients = CareRecipient.objects.filter(care_provider_location=self.location.id)
+        self.assertEqual(len(care_recipients), 0)
+
+    def test_admin_upload_binary_file(self):
+        file_path = os.path.join(os.path.dirname(__file__), "test_files/binary_file.bin")
+        with open(file_path, "rb") as file:
+            csv_file = SimpleUploadedFile("patients_test_data.csv", file.read(), content_type="text/csv")
+
+        response = self._get_upload_file_response(csv_file)
+        messages = self._convert_messages_to_str(response)
+        self.assertIn(CSVImportMessages.FILE_CORRUPTED_OR_BINARY.value, messages)
+        care_recipients = CareRecipient.objects.filter(care_provider_location=self.location.id)
+        self.assertEqual(len(care_recipients), 0)
+
+    def test_admin_upload_csv_file_with_invalid_columns(self):
+        file_path = os.path.join(os.path.dirname(__file__), "test_files/patients_invalid_column_set_test_data.csv")
+        with open(file_path, "rb") as file:
+            csv_file = SimpleUploadedFile("patients_test_data.csv", file.read(), content_type="text/csv")
+
+        response = self._get_upload_file_response(csv_file)
+        messages = self._convert_messages_to_str(response)
+        self.assertIn(CSVImportMessages.INVALID_COLUMN_SET.value, messages)
+        care_recipients = CareRecipient.objects.filter(care_provider_location=self.location.id)
```

Comment:
No need for the `filter` here, see above

Reply:
Agreed.